custom/reach/views.py

```diff
@@ -0,0 +1,142 @@
+from __future__ import absolute_import
+from __future__ import unicode_literals
+
+from django.http import JsonResponse
+from django.views.decorators.csrf import csrf_exempt
+
+from corehq.apps.domain.decorators import login_and_domain_required
+from corehq.apps.locations.models import SQLLocation
+from corehq.apps.locations.permissions import location_safe
+from django.utils.decorators import method_decorator
+from django.views.generic.base import TemplateView, View
+
+from custom.reach.const import INDICATOR_LIST, NUMERIC, PERCENT, COLORS
+
+
+class ReachDashboardView(TemplateView):
+    @property
+    def domain(self):
+        return self.kwargs['domain']
+
+    @property
+    def couch_user(self):
+        return self.request.couch_user
+
+    def get_context_data(self, **kwargs):
+        kwargs['domain'] = self.domain
+        # TODO add logic for user role type possible options should be MoHFW or MWCD
+        kwargs['user_role_type'] = 'MoHFW'
+        user_location = self.couch_user.get_sql_locations(self.domain).first()
+        kwargs['user_location_id'] = user_location.location_id if user_location else None
+
+        user_locations_with_parents = SQLLocation.objects.get_queryset_ancestors(
+            user_location, include_self=True
+        ).distinct() if user_location else []
+        parent_ids = [loc.location_id for loc in user_locations_with_parents]
+        kwargs['user_location_ids'] = parent_ids
+        return super(ReachDashboardView, self).get_context_data(**kwargs)
+
+
+@location_safe
+@method_decorator([login_and_domain_required], name='dispatch')
+class ProgramOverviewReport(ReachDashboardView):
+    template_name = 'reach/reports/program_overview.html'
+
+
+@location_safe
+@method_decorator([login_and_domain_required, csrf_exempt], name='dispatch')
+class ProgramOverviewReportAPI(View):
+    def post(self, request, *args, **kwargs):
+        # TODO add query to database
+        selected_month = int(self.request.POST.get('selectedMonth'))
+        selected_year = int(self.request.POST.get('selectedYear'))
+        selected_location = self.request.POST.get('selectedLocation')
+        return JsonResponse(data={'data': [
+            [
+                {
+                    'indicator': INDICATOR_LIST['registered_eligible_couples'],
+                    'format': NUMERIC,
+                    'color': COLORS['violet'],
+                    'numerator': 71682,
+                    'denominator': 140098,
+                    'past_month_numerator': 69354,
+                    'past_month_denominator': 130098
```

Comment:
Instead of `past_month_numerator` and `past_month_denominator`, I believe that this should just be `past_month_value` since there is no numerator or denominator

Reply:
:+1: